### DISCUSSION TITLE
When there are more managed resource resources, Composite become redundant and difficult to read

### DISCUSSION THREAD

**Discussion by zerunhu** [crossplane/crossplane#3720] [Q&A]
For example, my payment application might need ec2, security groups, CLB, mysql etc. I want to use a composite to combine these resources into a new Kind called "payApplication", which will automatically create all the managed resources I need as soon as I create the payApplication. But when I have a lot of managed resources, the composite file is redundant.

I used to use terraform, which can do the same thing with modules, but because modules can divide resources into multiple files, the structure of the module looks more obvious, whereas a composite would gather all the resources into one file, resulting in a very large and unobservable file

Is there a better way if I need to combine a series of managed resources?

**Comment by bobh66** [comment]
You can use nested composite resources to achieve the same effect as terraform modules.  You might have one composite for networking, one for the EKS cluster, one for nodegroup definition, one for supporting services and one for your application.  Then you can combine those into another composite that bundles them all together.

**Reply by ytsarev** [reply]
Thanks, @bobh66 for the great answer! https://github.com/upbound/platform-ref-aws is a good example of this approach.
